src/coreComponents/physicsSolvers/multiphysics/CoupledReservoirAndWellsBase.hpp

```diff
@@ -191,12 +192,84 @@ class CoupledReservoirAndWellsBase : public CoupledSolver< RESERVOIR_SOLVER, WEL
 
     DomainPartition & domain = this->template getGroupByPath< DomainPartition >( "/Problem/domain" );
 
-    // Validate well perforations: Ensure that each perforation is in a region targetted by the solver
+    // Validate well perforations: Ensure that each perforation is in a region targeted by the solver
     if( !validateWellPerforations( domain ))
     {
       return;
     }
+  }
+
+  virtual void
+  implicitStepSetup( real64 const & time_n,
+                     real64 const & dt,
+                     DomainPartition & domain ) override
+  {
+    Base::implicitStepSetup( time_n, dt, domain );
+
+    // we delay the computation of the transmissibility until the last minute
+    // because we want to make sure that the permeability has been updated (in the flow solver)
+    // this is necessary for some permeability models (like Karman-Kozeny) that do not use the imported permeability
+    // ultimately, we may want to use this mechanism to update the well transmissibility at each time step (if needed)
+    if( !m_isWellTransmissibilityComputed )
```

Comment:
Is it ever possible for this condition to be silently skipped?  Do we need to check the flag value at some point to confirm true?

Reply:
Yes, good idea, I added this check before assembly
https://github.com/GEOS-DEV/GEOS/blob/f43f5dd3211ceb468d5ce51e0d4cf57b350de002/src/coreComponents/physicsSolvers/multiphysics/CompositionalMultiphaseReservoirAndWells.cpp#L282-L285
to make sure that the computation is not silently ignored